Assets/Scripts/Room/Chat/InRoomChat.cs

```diff
@@ -1,593 +1,229 @@
-using Assets.Scripts.Characters.Titan;
-using Assets.Scripts.Gamemode.Options;
-using ExitGames.Client.Photon;
-using System;
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.UI;
+using static UnityEngine.Event;
+using static FengGameManagerMKII;
+using static ChatUtility;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Text;
+using System.Net;
 
 public class InRoomChat : Photon.MonoBehaviour
 {
-    private bool AlignBottom = true;
+    private const int MaxStoredMessages = 100;
+    private const int MaxMessageLength = 1000;
     public static readonly string ChatRPC = "Chat";
     private string inputLine = string.Empty;
     public bool IsVisible = true;
-    public static List<string> messages = new List<string>();
-    private Vector2 scrollPos = Vector2.zero;
+    private readonly List<string> messages = new List<string>();
     public InputField ChatInputField;
     public Text ChatText;
 
-    public void addLINE(string newLine)
+    public void OnGUI()
```

Comment:
Is this the correct Unity Message to use?
https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnGUI.html

Reply:
I have no idea.

Reply:
```suggestion
    private void Update()
```
> OnGUI is the only function that can implement the "Immediate Mode" GUI (IMGUI) system for rendering and handling GUI events. Your OnGUI implementation might be called several times per frame (one call per event).